src/compactor.rs

```diff
@@ -240,6 +240,22 @@ impl CompactorEventHandler {
     }
 
     fn write_manifest(&mut self) -> Result<(), SlateDBError> {
+        // write the checkpoint first so that it points to the manifest with the ssts
+        // being removed
+        self.tokio_handle.block_on(self.manifest.write_checkpoint(
+            None,
+            &CheckpointOptions {
+                // TODO(rohan): for now, just write a checkpoint with 15-minute expiry
+                //              so that it's extremely unlikely for the gc to delete ssts
+                //              out from underneath the writer. In a follow up, we'll write
+                //              a checkpoint with no expiry and with metadata indicating its
+                //              a compactor checkpoint. Then, the gc will delete the checkpoint
+                //              based on a configurable timeout
+                lifetime: Some(Duration::from_secs(900)),
```

Comment:
Just confirming: it looks like the default compaction poll interval is 5s. Worst case, on default, we'd be making a new checkpoint every 5s and carry ~180 checkpoints in the manifest for the compactor. I think that's OK since most of the time, poll should be a noop and not result in a manifest write.

Reply:
Yeah that's pretty much right. We only write the manifest after completing a compaction, so it should only be leaving checkpoints behind when they're needed.